### DISCUSSION TITLE
Proptesting & Fuzzing

### DISCUSSION THREAD

**Discussion by JackYoustra** [pointfreeco/swift-composable-architecture#1531] [General]
I notice in the given examples that test cases are often hardcoded, static examples. Has anyone tried testing relations with fuzzing and proptesting, perhaps with a package like [SwiftCheck](https://github.com/typelift/SwiftCheck)?

**Comment by mbrandonw** [comment]
I have not personally experimented with this, but you might be interested in this [talk](https://www.youtube.com/watch?v=23_qZePMQjA) from Loïc Lecrenier at the last ever Functional Swift Conference. He discusses applying fuzz testing to an Elm application.

We would love to hear of any experiments people do!

**Reply by JackYoustra** [reply]
Thanks for the libfuzzer talk! Was looking for something like this.

**Reply by stephencelis** [reply]
If you do get something working with TCA we'd love to hear about it!
